iceoryx_hoofs/test/moduletests/test_shared_memory.cpp

```diff
@@ -19,77 +19,81 @@
 
 #include "iceoryx_hoofs/internal/posix_wrapper/shared_memory_object/shared_memory.hpp"
 #include "iceoryx_hoofs/platform/stat.hpp"
+#include "iceoryx_hoofs/posix_wrapper/posix_call.hpp"
+
+#include <fcntl.h>
 
 namespace
 {
 using namespace testing;
+using namespace iox;
+using namespace iox::posix;
 
 class SharedMemory_Test : public Test
 {
   public:
     void SetUp() override
     {
-        internal::CaptureStderr();
+        testing::internal::CaptureStderr();
     }
 
     void TearDown() override
     {
-        std::string output = internal::GetCapturedStderr();
+        std::string output = testing::internal::GetCapturedStderr();
         if (Test::HasFailure())
         {
             std::cout << output << std::endl;
         }
     }
-};
 
-TEST_F(SharedMemory_Test, CTorWithValidArguments)
-{
-    auto sut = iox::posix::SharedMemory::create("/ignatz",
+    static constexpr const char SUT_SHM_NAME[] = "/ignatz";
+
+    iox::cxx::expected<iox::posix::SharedMemory, iox::posix::SharedMemoryError>
+    createSharedMemory(const iox::posix::SharedMemory::Name_t& name, const iox::posix::OpenMode openMode)
+    {
+        return iox::posix::SharedMemory::create(name,
                                                 iox::posix::AccessMode::READ_WRITE,
-                                                iox::posix::OwnerShip::MINE,
+                                                openMode,
                                                 S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP | S_IROTH | S_IWOTH,
                                                 128);
+    }
+
+    bool createRawSharedMemory(const iox::posix::SharedMemory::Name_t& name)
+    {
+        return !iox::posix::posixCall(iox_shm_open)(
```

Comment:
Let's assume I am a chicken, should we call `unlinkIfExists` first to ensure this call will always succeed?

I also got confused by the names of the two `create...` functions since they are quite similar. Should `createSharedMemory` maybe be renamed to `createSut` or similar?

Reply:
Explicitly no since some tests require that the shared memory already exist and this will fail like `ExclusiveCreateFailsWhenShmExists`